docs/src/implementation.md

```diff
@@ -2,25 +2,12 @@
 
 `CategoricalArray` is made of the two fields:
 
-- `refs`: an integer array that stores the position of the category level in the `index` field of `CategoricalPool` for each `CategoricalArray` element; `0` denotes a missing value (for `CategoricalArray{Union{T, Missing}}` only).
+- `refs`: an integer array that stores the position of the category level in the `levels` field of `CategoricalPool` for each `CategoricalArray` element; `0` denotes a missing value (for `CategoricalArray{Union{T, Missing}}` only).
```

Comment:
might be good to clarify if 0 can denote `missing` _only_ for `Union{T, Missing}` CategoricalArrays; like, if it's just a regular `CategoricalVector{String}`, would 0 point to the first level? Or is 0 exclusively for `missing` always.

Reply:
For other arrays it corresponds to an undefined entry. But can we leave unrelated improvements to another PR? This one is already messy enough.